failsafe/failsafe.py

```diff
@@ -75,7 +77,9 @@ def __init__(self, retry_policy=None, circuit_breaker=None):
             except Exception as e:
                 context.errors += 1
                 recent_exception = e
-                retry = self.retry_policy.should_retry(context, e)
+                retry = self.exception_handling_policy.should_retry(context, e)
+                if self.exception_handling_policy.should_raise(e):
```

Comment:
I'm not sure about this. If an exception is in both retriable_exceptions and abortable_exceptions what should happen?
My guess is that we should retry first and then, after the retries have been exhausted, we should raise without reporting a failure.